### ISSUE TITLE
Enhancement: [require-await] Add suggestion to remove async keyword

### ISSUE THREAD

**Issue by reduckted** [typescript-eslint/typescript-eslint#9621]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/require-await

### Description

#6509 proposed adding an automatic fixer for the `require-await` rule, however that was rightly rejected because automatically applying a fix could change the type of the function and therefore break code.

A better option would be to add a [_suggestion_](https://eslint.org/docs/latest/extend/custom-rules#providing-suggestions) so that the problem can be semi-automatically fixed, _but only when you ask the editor_ to apply the suggestion.

### Fail

```TypeScript
const arrow = async () => {
    return;
};
```

### Pass

```TypeScript
const arrow = () => {
    return;
};
```

**Comment by Josh-Cena** [comment]
Hi, this rule is a core rule extension. Any behavior that might be implemented by the base rule should be implemented by the base rule before we consider it. In this case, it's definitely fine for the base rule to add said suggestion, so I suggest you propose it to ESLint first.

**Comment by kirkwaiblinger** [comment]
+1 to @Josh-Cena's comment. Closing this for now, until/unless a proposal is successfully made to ESLint (https://github.com/eslint/eslint/).

**Comment by JoshuaKGoldberg** [comment]
https://github.com/eslint/eslint/issues/18713 is merged -> we're good to go! 🚀
